crates/ty_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -1118,7 +1118,41 @@ impl<'db> UseDefMapBuilder<'db> {
             .add_or_constraint(self.reachability, snapshot.reachability);
     }
 
+    fn mark_reachability_constraints(&mut self) {
+        // We only walk the fields that are copied through to the UseDefMap when we finish building
+        // it.
+        for bindings in &self.bindings_by_use {
+            bindings.mark_reachability_constraints(&mut self.reachability_constraints);
+        }
```

Comment:
I haven't looked closely so there might be a very obvious reason why you decided not to do that.

Would it be possible to mark those constraints as used as we add bindings, place_state, constraint etc to avoid the extra traversals?

Reply:
Makes sense. I'm fine keeping it this way for now